### PR TITLE
KBucket clean up. V3-923

### PR DESCRIPTION
**Pull request by aligeti**
https://storjlabs.atlassian.net/browse/V3-923

### CODE REVIEW

File: pkg/kademlia/routing.go [hunk 1]
```diff
@@ -91,36 +91,19 @@ func (rt *RoutingTable) CacheSize() int {
 
 // GetBucket retrieves the corresponding kbucket from node id
 // Note: id doesn't need to be stored at time of search
-func (rt *RoutingTable) GetBucket(id storj.NodeID) (bucket dht.Bucket, ok bool) {
+func (rt *RoutingTable) GetBucket(id storj.NodeID) ([]*pb.Node, bool) {
```
Comment (coyle) [inline pkg/kademlia/routing.go]: Why change this to return a `[]*pb.Node` instead of a Bucket type?

Comment (aligeti) [reply pkg/kademlia/routing.go]: the ticket is to clean up the bucket interface as it is not used

Comment (coyle) [reply pkg/kademlia/routing.go]: No, the bucket interface is still around and being used. The implementation at pkg/kademlia/bucket.go was not being used

Comment (coyle) [reply pkg/kademlia/routing.go]: https://github.com/storj/storj/pull/898/files#diff-10d55b8cd7e4b1ce8e2c34ca4f5d2ea9L49

Comment (coyle) [reply pkg/kademlia/routing.go]: I'm ok with getting rid of the interface but it's still in the code and `GetBucket` seems like it should return a bucket type. Maybe change the name to be `GetNodes` ?

File: pkg/kademlia/routing.go [hunk 2]
```diff
@@ -89,38 +89,21 @@ func (rt *RoutingTable) CacheSize() int {
 	return rt.rcBucketSize
 }
 
-// GetBucket retrieves the corresponding kbucket from node id
+// GetNodes retrieves the corresponding kbucket from node id
```
Comment (jenlij) [inline pkg/kademlia/routing.go]: maybe update the comment here to say "retrieves nodes within the same kbucket as the given node id"?

File: pkg/kademlia/routing_test.go [hunk 3]
```diff
@@ -47,6 +47,10 @@ func TestGetBucket(t *testing.T) {
 	assert.True(t, ok)
 	assert.NoError(t, err)
 
+	type KBucket struct {
```
Comment (jenlij) [inline pkg/kademlia/routing_test.go]: do we still need this struct?

File: pkg/kademlia/routing.go (hunk 2 above)
Comment (aligeti) [reply pkg/kademlia/routing.go]: yes. Made the change

File: pkg/kademlia/routing_test.go (hunk 3 above)
Comment (aligeti) [reply pkg/kademlia/routing_test.go]: Done..

**Pull request merged**